### PR TITLE
Update file-manager extension

### PR DESCRIPTION
**Pull request by BalliAsghar**
## Changes
- **Update packages:** Upgraded dependencies.
- **Remove deprecated code**: Replaced old code for compatibility with latest @raycast/api.

## Features
- **Move files to trash**: Added feature to move files to trash.
- **Delete files**: Added feature to permanently delete files.
- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

**Comment by es183923** [comment]
I'm not sure that the delete forever option is a good idea. I think just having the move to trash option is enough.

**Comment by BalliAsghar** [comment]
> I'm not sure that the delete forever option is a good idea. I think just having the move to trash option is enough.

I understand your concern. In the updated implementation, the Delete action is optional and controlled by the showDeleteOption preference. By setting it to false, only the "Move to Trash" option will be available. Users who want the Delete action can enable it by setting showDeleteOption to true. This ensures a safer default while still providing the option for those who need it.

File: extensions/file-manager/src/utils.tsx [hunk 1]
```diff
@@ -38,8 +37,59 @@ export type PreferencesType = {
...
+        refresh();
+        showToast(Toast.Style.Success, "File Deleted", `${fileName}`);
+      },
+    },
+  };
+
+  await confirmAlert(options);
+
+  return;
+}
```
Comment (thomaslombart) [inline extensions/file-manager/src/utils.tsx:62]: Given the delete actions are permanent, it'd be better to let users know what they're doing is irreversible.

```suggestion
export async function deleteFile(filePath: string, fileName: string, refresh: () => void) {
  const options: Alert.Options = {
    title: "Permanently Delete File?",
    message: `Are you sure you want to delete ${fileName}? This action cannot be undone.`,
    icon: Icon.Eraser,
    primaryAction: {
      title: "Delete",
      style: Alert.ActionStyle.Destructive,
      onAction: async () => {
        await runShellScript(`rm -rf "${filePath}"`);
        refresh();
        showToast(Toast.Style.Success, "File Deleted", `${fileName}`);
      },
    },
  };

  await confirmAlert(options);

  return;
}
```

File: extensions/file-manager/src/utils.tsx [hunk 2]
```diff
@@ -38,8 +37,59 @@ export type PreferencesType = {
...
+        refresh();
+        showToast(Toast.Style.Success, "Directory Deleted", `${folderName}`);
+      },
+    },
+  };
+
+  await confirmAlert(options);
+
+  return;
+}
```
Comment (thomaslombart) [inline extensions/file-manager/src/utils.tsx:83]:
```suggestion
export async function deleteDirectory(folderPath: string, folderName: string, refresh: () => void) {
  const options: Alert.Options = {
    title: "Permanently Delete Directory?",
    message: `Are you sure you want to delete ${folderName}? This action cannot be undone.`,
    icon: Icon.Eraser,
    primaryAction: {
      title: "Delete",
      style: Alert.ActionStyle.Destructive,
      onAction: async () => {
        await runShellScript(`rm -rf "${folderPath}"`);
        refresh();
        showToast(Toast.Style.Success, "Directory Deleted", `${folderName}`);
      },
    },
  };

  await confirmAlert(options);

  return;
}
```

File: extensions/file-manager/src/utils.tsx [hunk 3]
```diff
@@ -38,8 +37,59 @@ export type PreferencesType = {
   showFilePermissions: boolean;
   showFileSize: boolean;
   startDirectory: string;
+  showDeleteOption: boolean;
```
Comment (thomaslombart) [inline extensions/file-manager/src/utils.tsx:40]:
```suggestion
  showDeleteActions: boolean;
```

File: extensions/file-manager/src/utils.tsx [hunk 4]
```diff
@@ -62,20 +112,39 @@ export function DirectoryItem(props: { fileData: FileDataType }) {
...
+            title="Move to Trash"
+            shortcut={{ modifiers: ["cmd"], key: "t" }}
+            paths={filePath}
+            onTrash={() => {
+              showToast(Toast.Style.Success, "Moved to Trash", `Directory: ${filePath}`);
+              props.refresh();
+            }}
+          />
+
+          {preferences.showDeleteOption && (
```
Comment (thomaslombart) [inline extensions/file-manager/src/utils.tsx:132]:
```suggestion
          {preferences.showDeleteActions && (
```

File: extensions/file-manager/src/utils.tsx [hunk 5]
```diff
@@ -85,26 +154,47 @@ export function FileItem(props: { fileData: FileDataType }) {
...
+          <Action.Trash
+            title="Move to Trash"
+            shortcut={{ modifiers: ["cmd"], key: "t" }}
+            paths={filePath}
+            onTrash={() => {
+              showToast(Toast.Style.Success, "Moved to Trash", `File: ${filePath}`);
+              props.refresh();
+            }}
+          />
+          {preferences.showDeleteOption && (
```
Comment (thomaslombart) [inline extensions/file-manager/src/utils.tsx:182]:
```suggestion
          {preferences.showDeleteActions && (
```

File: extensions/file-manager/src/utils.tsx [hunk 6]
```diff
@@ -148,35 +247,44 @@ export function SymlinkItem(props: { fileData: FileDataType }) {
...
               content={filePath}
               shortcut={{ modifiers: ["cmd", "shift"], key: "c" }}
             />
-            <CopyToClipboardAction
+            <Action.CopyToClipboard
               title="Copy Original File Path"
               content={originalPath}
               shortcut={{ modifiers: ["cmd", "opt"], key: "c" }}
             />
+            {preferences.showDeleteOption && (
```
Comment (thomaslombart) [inline extensions/file-manager/src/utils.tsx:262]:
```suggestion
            {preferences.showDeleteActions && (
```

File: extensions/file-manager/src/utils.tsx [hunk 7]
```diff
@@ -119,22 +209,31 @@ export function SymlinkItem(props: { fileData: FileDataType }) {
...
               content={filePath}
               shortcut={{ modifiers: ["cmd", "shift"], key: "c" }}
             />
-            <CopyToClipboardAction
+            <Action.CopyToClipboard
               title="Copy Original Directory Path"
               content={filePath}
               shortcut={{ modifiers: ["cmd", "opt"], key: "c" }}
             />
+            {preferences.showDeleteOption && (
```
Comment (thomaslombart) [inline extensions/file-manager/src/utils.tsx:228]:
```suggestion
            {preferences.showDeleteActions && (
```

**Review by thomaslombart — CHANGES_REQUESTED** [review]
Thanks for this PR. I've reviewed it and have some feedback.

Please, request a new review when you are ready. Feel free to contact me here or in the [Slack community](https://www.raycast.com/community) if you have any questions.Thanks for this PR 😊
I've reviewed it and have some feedback:



Please, request a new review when you are ready. Feel free to contact me here or in the [Slack community](https://www.raycast.com/community) if you have any questions.

**Review by thomaslombart — APPROVED** [review]
Looks good to go! 🚀

**Pull request merged**